src/ssl.c

```diff
@@ -35907,7 +35907,7 @@ int wolfSSL_BN_is_word(const WOLFSSL_BIGNUM* bn, WOLFSSL_BN_ULONG w)
         return WOLFSSL_FAILURE;
     }
 
-    if (w <= (WOLFSSL_BN_ULONG)MP_MASK) {
+    if ((sizeof(w) <= sizeof(MP_MASK)) || (w <= (WOLFSSL_BN_ULONG)MP_MASK)) {
```

Comment:
These should just be `if ((mp_digit)w <= MP_MASK) {`

Reply:
@dgarske , That doesn't work (tested it) because it's a comparison of the values, and in situations where the types of both `mp_digit` (or `WOLFSSL_BN_ULONG`) and `MP_MASK` are the same (i.e. unsigned types of the same bit-length) the condition is always TRUE, which triggers the IAR tools warning.

For example, with `mp_digit` (or `WOLFSSL_BN_ULONG`) being a 32-bit unsigned value, and `MP_MASK` as `0xffffffffU`, the condition check becomes:
```
if ([0..0xffffffff] <= 0xffffffff)
```
...which is always TRUE.  The original condition aimed to check whether the value of `w` is less than `MP_MASK` (which is a maximum value), and that only makes sense when the bit-length of `w` is larger than that of `MP_MASK`, making it *possible* for the value of `w` to be larger than `MP_MASK`.  

Casting doesn't fool the (IAR) compiler, either... both can be cast to large types and compiler still sees the problem (i.e. it knows the range of values `w` can be with or without the cast).  The following causes the warning, too:
```
 if ((unsigned long long)w <= (unsigned long long)MP_MASK) {
```
Hmm... Since the value/length of `MP_MASK` is configured based on a number of settings values (math library used, SP_WORD_SIZE, etc.) it might be possible to add conditional-compilation around the check itself to detect at compile-time whether the check is needed, based on the system configuration... I hadn't explored that approach.  In the current, "run-time-size-check" approach, my guess is that the compiler is smart enough to know at compile-time whether the run-time size check is needed based on the types involved (it's smart enough to throw the warning!) and optimize it out, but that's only a guess... If I can get conditional-compilation checks to do the size-check, that would ensure that the condition is only included in the compile when it's needed (avoiding the warning)... 

Please let me know if I should pursue the conditional-compilation / compile-time check approach!

Reply:
Ideally mp_digit and WOLFSSL_BN_ULONG would be same size but they aren’t. Your original fix is fine with me but is very confusing when reviewing the code. Consider adding a code comment.

Reply:
Agreed, I added comments to the modified conditions.  Please let me know if they are not sufficient or still obtuse and I can expand or reword them.   Thanks!